timeseries/src/autogluon/timeseries/transforms/scaler.py

```diff
@@ -0,0 +1,121 @@
+        return self.fit(data=data).transform(data=data)
+
+    def fit(self, data: TimeSeriesDataFrame) -> "LocalTargetScaler":
+        target_series = data[self.target].replace([np.inf, -np.inf], np.nan)
+        self.loc, self.scale = self._compute_loc_scale(target_series)
+        if self.loc is not None:
+            self.loc = self.loc.replace([np.inf, -np.inf], np.nan).fillna(0.0)
+        if self.scale is not None:
+            self.scale = self.scale.clip(lower=self.min_scale).replace([np.inf, -np.inf], np.nan).fillna(1.0)
+        return self
+
+    def _reindex_loc_scale(self, item_index: pd.Index) -> Tuple[Union[np.ndarray, float], Union[np.ndarray, float]]:
+        """Reindex loc and scale parameters for the given item_ids and convert them to an array-like."""
+        if self.loc is not None:
+            loc = self.loc.reindex(item_index).to_numpy()
+        else:
+            loc = 0.0
+        if self.scale is not None:
+            scale = self.scale.reindex(item_index).to_numpy()
+        else:
+            scale = 1.0
+        return loc, scale
+
+    def transform(self, data: TimeSeriesDataFrame) -> TimeSeriesDataFrame:
+        """Apply scaling to the target column in the dataframe."""
+        loc, scale = self._reindex_loc_scale(item_index=data.index.get_level_values(ITEMID))
+        return data.assign(**{self.target: (data[self.target] - loc) / scale})
+
+    def inverse_transform(self, predictions: TimeSeriesDataFrame) -> TimeSeriesDataFrame:
+        """Apply inverse scaling to all columns in the predictions dataframe."""
+        loc, scale = self._reindex_loc_scale(item_index=predictions.index.get_level_values(ITEMID))
+        return predictions.assign(**{col: predictions[col] * scale + loc for col in predictions.columns})
+
+
+class LocalStandardScaler(LocalTargetScaler):
+    """Applies standard scaling to each time series in the dataset."""
+
+    def _compute_loc_scale(self, target_series: pd.Series) -> Tuple[pd.Series, pd.Series]:
+        stats = target_series.groupby(level=ITEMID, sort=False).agg(["mean", "std"])
+        return stats["mean"], stats["std"]
+
+
+class LocalMeanAbsScaler(LocalTargetScaler):
+    """Applies mean absolute scaling to each time series in the dataset."""
+
+    def _compute_loc_scale(self, target_series: pd.Series) -> Tuple[pd.Series, pd.Series]:
+        scale = target_series.abs().groupby(level=ITEMID, sort=False).agg("mean")
+        return None, scale
+
+
+class LocalMinMaxScaler(LocalTargetScaler):
+    """Applies min/max scaling to each time series in the dataset."""
+
+    def _compute_loc_scale(self, target_series: pd.Series) -> Tuple[pd.Series, pd.Series]:
+        stats = target_series.abs().groupby(level=ITEMID, sort=False).agg(["min", "max"])
+        scale = (stats["max"] - stats["min"]).clip(lower=self.min_scale)
+        loc = stats["min"] / scale
+        return loc, scale
+
+
+class LocalRobustScaler(LocalTargetScaler):
```

Comment:
All scalers above have a docstring. Is there a short one-line description that could add some information here?

Reply:
Good idea, added some docstrings.